### PR TITLE
chore: update contributing guidelines

### PR DESCRIPTION
**Pull request by wmertens**
Update the guidelines now that we'll have only squash and merge merges.

[Rendered](https://github.com/QwikDev/qwik/blob/327eb5b2e8c97a6674f0118bc7faad6719195101/CONTRIBUTING.md)

### CODE REVIEW

**Review by gioboa — APPROVED** [review]
Nice updates @wmertens

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,14 +8,21 @@ b) Framework functionality achievable only by the core.
...
-- [ ] Feature / enhancement
-- [ ] Bug
-- [ ] Docs / tests / types / typos
-- [ ] Infra
+<!-- pick one and remove the others -->
+
+- Feature / enhancement
+- Bug
+- Docs / tests / types / typos
+- Infra
```
Comment (maiieul) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: nit pick but I prefer the checkbox, a little more cumbersome to delete 3 lines 🙈

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -24,10 +31,15 @@ If you believe your functionality is valuable to the entire Qwik Community, disc
...
+* add items that are relevant and need to be done before merging
+-->
 
 - [ ] My code follows the [developer guidelines of this project](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md)
-- [ ] I have performed a self-review of my own code
-- [ ] I have ran `pnpm change` and documented my changes
-- [ ] I have made corresponding changes to the Qwik docs
+- [ ] Performed a self-review of my own code
+- [ ] Added changesets with `pnpm change`
+- [ ] Made corresponding changes to the Qwik docs
```
Comment (maiieul) [inline .github/PULL_REQUEST_TEMPLATE.md:44]: previous wording was clearer to me. It's mostly for people that are not contributing often so better be clear

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (wmertens) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: But then we see all these PRs that are "incomplete" in the overview

**Review by mhevery — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (maiieul) [reply .github/PULL_REQUEST_TEMPLATE.md:44]: yes 👍

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (maiieul) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: what do you mean? screenshot maybe?

Comment (wmertens) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: image

Comment (wmertens) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: It will also make the description less busy. Let's try it and see, we can still change it later.

**Review by shairez — APPROVED** [review]
Great job @wmertens ! 💪

**Pull request merged**